src/components/BoardGenerator/BoardGenerator.js

```diff
@@ -0,0 +1,29 @@
+import { v4 as uuidv4 } from 'uuid';
+import { animals, frozen } from '../../componentAssets/Images';
+
+const BoardGenerator = (cardBack, boardSize) => {
+  const cardBackImg = cardBack === 'animals' ? animals : frozen;
+  const createArray = (boardSize) => {
+    const animalsObj = Object.keys(cardBackImg);
+    const singleArray = Array.from({ length: boardSize }, () => ({}));
+    singleArray.forEach((el, idx) => {
+      el.imgSrc = cardBackImg[animalsObj[idx]];
+      el.pairId = idx;
+      el.isVisible = false;
+    });
+
+    const singleArrayCopy = JSON.parse(JSON.stringify(singleArray));
+
+    const shuffledArray = [...singleArray, ...singleArrayCopy].sort(
```

Comment:
As far as I know concat is doing shallow copy and copies objects and arrays by reference instead of by value, which is key feature in the next step where I have to add unique 'id' values to every array element.